docs/developer/branching-and-workflow.md

```diff
@@ -0,0 +1,30 @@
+# Branching and Workflow
+
+NKG is an open source and public repository; our goal is to keep the number of branches in the repository to a minimum:
+the main branch, release branches and long-term feature branches.
+
+Internal developers and external contributors will follow a fork and merge process. Each contributor should fork the
+repo to their own space; branch, experiment, develop and prepare a PR to merge their work into NKG’s main branch. This
+way ephemeral developer branches will remain outside the main repository.
+
+Below is an example of following the merge and fork process. Developer Alice:
+
+- Forks `github.com/nginxinc/nginx-kubernetes-gateway` → `github.com/<alice-user-id>/nginx-kubernetes-gateway`
+- Alice develops a feature or bugfix - using as many ephemeral branches as she needs.
+- Alice creates a
+  PR `github.com/<alice-user-id>/nginx-kubernetes-gateway:feature/some-feature` → `github.com/nginxinc/nginx-kubernetes-gateway:main`
```

Comment:
It's probably worth mentioning how to keep your fork up to date. There are probably multiple ways to do this, but personally, after cloning my fork, I add the upstream:

```
$ git remote add upstream git@github.com:nginxinc/nginx-kubernetes-gateway.git
```

and now have two remotes configured:

```
$ git remote -v
origin	git@github.com:sjberman/nginx-kubernetes-gateway.git (fetch)
origin	git@github.com:sjberman/nginx-kubernetes-gateway.git (push)
upstream	git@github.com:nginxinc/nginx-kubernetes-gateway.git (fetch)
upstream	git@github.com:nginxinc/nginx-kubernetes-gateway.git (push)
```

I can then `git pull upstream main` to keep my local main branch in sync with the main repo.

Maybe there are simpler ways, but this is what I've always done.

Reply:
Added. Let me know what you think

Reply:
LGTM